.github/workflows/ci.yml

```diff
@@ -0,0 +1,159 @@
+name: ci
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  RUST_PREVIOUS_VERSION: 1.86.0
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  # Enable consistent use of `RUST_PREVIOUS_VERSION` across different jobs
+  prepare-checks:
+    runs-on: ubuntu-latest
+
+    outputs:
+      rust-prev-version: ${{ env.RUST_PREVIOUS_VERSION }}
+
+    steps:
+      - name: Set rust-prev-version output
+        run: echo "rust-prev-version=${RUST_PREVIOUS_VERSION}" >> $GITHUB_OUTPUT
+
+################################## FORMAT AND LINTS LAYER ######################
+
+  clippy-rustfmt:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+          components: clippy, rustfmt
+
+      - name: Run cargo clippy
+        run: cargo clippy --all-targets -- -D warnings
+
+      - name: Run rustfmt
+        run: cargo fmt --all -- --check --verbose
+
+################################## BUILD AND DOCS LAYER ########################
+
+  build:
+    needs: [clippy-rustfmt]
+
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+
+    runs-on: ${{ matrix.platform }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Build
+        run: cargo build --verbose
```

Comment:
Let's use `cargo test` with a combination of different features instead of `cargo build`. The coverage tests on `ubuntu`, but it does not consider tests on other platforms. 

Add more steps considering the features, for example, we have defined two features in  `ascot` crate which activates different tests. It would be better to analyse all of them.